feat(gpu-operator): initial gpu-operator

jakepage91 opened the pull request:
Added and tested GPU-Operator and Node Feature Discovery (which is a dependency)

![Screenshot 2024-07-26 at 13 42 35](https://github.com/user-attachments/assets/6bc68996-dfb3-45aa-88cc-f2a44ee3e72c)

kosmoz commented on packages/gpu-operator/v24.3.0+1/package.yaml:
```diff
@@ -0,0 +1,20 @@
+  - name: node-feature-discovery
+references:
+  - label: ArtifactHub
+    url: 
```
empty `url` is not allowed in the schema. If there is no artifacthub URL, please remove the entry

kosmoz commented on packages/gpu-operator/v24.3.0+1/package.yaml:
```diff
@@ -0,0 +1,20 @@
+    url: 
+  - label: GitHub
+    url: https://github.com/NVIDIA/gpu-operator
+  - label: website
```
```suggestion
  - label: Website
```

kosmoz commented on packages/gpu-operator/v24.3.0+1/package.yaml:
```diff
@@ -0,0 +1,20 @@
+
+name: "gpu-operator"
+shortDescription: >-
+  The GPU Operator allows administrators of Kubernetes clusters to manage GPU nodes just like CPU nodes in the cluster.
```
![image](https://github.com/user-attachments/assets/b86cbcbe-2816-4cea-8d61-57eebdcad3ef)

I think the short description is a little too long (the interesting part is truncated; see screenshot). Maybe something like this would fit better?

> Management of NVIDIA components needed to provision GPU

kosmoz reviewed (commented):
This package will install NFD twice (once as dependency, once as part of the helm chart). To disable the latter, the following `values` must be set in the helm config:

```yaml
nfd:
  enabled: false
```

Source: https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/getting-started.html#common-chart-customization-options

jakepage91 commented:
Addressed all comments @kosmoz and I also tested the deployment locally and it worked. I think we are good to go here. 

![Screenshot 2024-07-31 at 15 31 14](https://github.com/user-attachments/assets/ecde03c3-1c65-4e62-ae90-8b13b4482304)

kosmoz commented:
Cool, the example from the install guide worked!
```
─> echo '
    apiVersion: v1
    kind: Pod
    metadata:
      name: cuda-vectoradd
    spec:
      restartPolicy: OnFailure
      containers:
      - name: cuda-vectoradd
        image: "nvcr.io/nvidia/k8s/cuda-sample:vectoradd-cuda11.7.1-ubuntu20.04"
        resources:
          limits:
            nvidia.com/gpu: 1' | kubectl apply -f -
pod/cuda-vectoradd created
─> kubectl logs pod/cuda-vectoradd
[Vector addition of 50000 elements]
Copy input data from the host memory to the CUDA device
CUDA kernel launch with 196 blocks of 256 threads
Copy output data from the CUDA device to the host memory
Test PASSED
Done
```
